sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Queues/QueueListener.cs

```diff
@@ -444,185 +451,14 @@ internal static void RegisterSharedWatcherWithQueueProcessor(QueueProcessor queu
-                bool queueLengthIncreasing =
-                IsTrueForLastN(
-                    metrics,
-                    NumberOfSamplesToConsider,
-                    (prev, next) => prev.QueueLength < next.QueueLength);
-                if (queueLengthIncreasing)
-                {
-                    status.Vote = ScaleVote.ScaleOut;
-                    _logger.LogInformation($"Queue length is increasing for '{_queue.Name}'");
-                    return status;
-                }
-            }
-
-            if (metrics[0].QueueTime > TimeSpan.Zero && metrics[0].QueueTime < metrics[NumberOfSamplesToConsider - 1].QueueTime)
-            {
-                bool queueTimeIncreasing =
-                    IsTrueForLastN(
-                        metrics,
-                        NumberOfSamplesToConsider,
-                        (prev, next) => prev.QueueTime <= next.QueueTime);
-                if (queueTimeIncreasing)
-                {
-                    status.Vote = ScaleVote.ScaleOut;
-                    _logger.LogInformation($"Queue time is increasing for '{_queue.Name}'");
-                    return status;
-                }
-            }
-
-            bool queueLengthDecreasing =
-                IsTrueForLastN(
-                    metrics,
-                    NumberOfSamplesToConsider,
-                    (prev, next) => prev.QueueLength > next.QueueLength);
-            if (queueLengthDecreasing)
-            {
-                status.Vote = ScaleVote.ScaleIn;
-                _logger.LogInformation($"Queue length is decreasing for '{_queue.Name}'");
-                return status;
-            }
-
-            bool queueTimeDecreasing = IsTrueForLastN(
-                metrics,
-                NumberOfSamplesToConsider,
-                (prev, next) => prev.QueueTime > next.QueueTime);
-            if (queueTimeDecreasing)
-            {
-                status.Vote = ScaleVote.ScaleIn;
-                _logger.LogInformation($"Queue time is decreasing for '{_queue.Name}'");
-                return status;
-            }
-
-            _logger.LogInformation($"Queue '{_queue.Name}' is steady");
-
-            return status;
-        }
-
-        private static bool IsTrueForLastN(IList<QueueTriggerMetrics> samples, int count, Func<QueueTriggerMetrics, QueueTriggerMetrics, bool> predicate)
-        {
-            Debug.Assert(count > 1, "count must be greater than 1.");
-            Debug.Assert(count <= samples.Count, "count must be less than or equal to the list size.");
-
-            // Walks through the list from left to right starting at len(samples) - count.
-            for (int i = samples.Count - count; i < samples.Count - 1; i++)
-            {
-                if (!predicate(samples[i], samples[i + 1]))
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return _scaleMonitor != null ? _scaleMonitor.Value : null;
```

Comment:
`_scaleMonitor.Value` can not be null.

```suggestion
            return _scaleMonitor.Value;
```